src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -1035,7 +1026,14 @@ public int GetCookiesCount()
             }
         }
 
-        public object SyncRoot => _lock;
+        public object SyncRoot
+        {
+            get
+            {
+                Debug.Assert(_list != null);
```

Comment:
Consider adding a message to the assert highlighting that PathList should never be default initialized and only ever created with Create.